docs/examples.md

```diff
@@ -0,0 +1,170 @@
+    shortcode_name = "anchor_image"
+    
+
+    def pre_filter(self, string):
+        # This method is required and will be called to transform a specific
+        # HTML structure found by BeautifulSoup into a custom HTML tag
+        # This example would produce an HTML tag of <wagtail_block_anchor_image>
+        soup = BeautifulSoup(string, "html.parser")
+        anchor_tags = soup.find_all("a")
+
+        anchors = []
+        for anchor in anchor_tags:
+            if anchor.find("img"):
+                anchors.append(anchor)
+
+        for a in anchors:
+            attrs = {}
+            attrs["data-src"] = a.find("img").attrs["src"]
+            attrs["data-caption"] = a.find("img").attrs["title"]
+            attrs["data-class"] = a.find("img").attrs["class"]
+            attrs["data-href"] = a.attrs["href"]
+            replacement_custom_tag = soup.new_tag(self.element_name)
+            replacement_custom_tag.attrs = attrs
+            a.replace_with(replacement_custom_tag)
+
+        return str(soup)
+
+    def construct_block(self, soup):
+        # get_or_save_image() is not included in the shortcode handler
+        # it is included in the package and can be used if required
+        image_file = get_or_save_image(soup.attrs["data-src"])
+
+        if image_file:
+            # The block type as a dict is returned to the BlockBuilder
+            return {
+                "type": "image",
+                "value": {
+                    "image": image_file.id,
+                    "caption": soup.attrs.get("data-caption"),
+                    "alignment": soup.attrs.get("data-class"),
+                    "link": soup.attrs.get("data-href"),
+                },
+            }
+        else:
+            # You should return at least one valid block here.
+            # This returns a raw_html block containing the custom HTML tag.
+            # It's a way to handle content that for some reason could not be parsed.
+            return {
+                "type": "raw_html",
+                "value": str(soup),
+            }
+```
+
+Because the handler will add a custom HTML tag to the content the default action of the bleach_clean() filter needs to be altered.
+
+Specifically, it needs to know what the custom HTML tag name is and what are valid attributes of the tag.
+
+Then generated custom HTML tag for the example above would be:
+
+```html
+<wagtail_block_anchor_image 
+    data-href="http://www.example.com/some-content" 
+    data-class="alignleft" 
+    data-src="http;//www.example.com/image.jpg" 
+    data-caption="Image title">
+</wagtail_block_anchor_image>
+```
+
+Before the custom HTML tag above is converted to a Wagtail block all the content is run through bleach_clean() to remove any unwanted HTML tags. The default list of HTML tags and attributes doesn't include the custom HTML here.
+
+### Add extra configuration to allow this HTML tag is pass through bleach_clean()
```

Comment:
**Suggestion**: Correct this typo "tag is pass through". I'm not sure what was originally intended.